packages/combobox/src/LionCombobox.js

```diff
@@ -335,8 +340,12 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
    */
   __unsyncCheckedIndexOnInputChange() {
     const autoselect = this._autoSelectCondition();
-    if (!this.multipleChoice && !autoselect && !this._inputNode.value.startsWith(this.modelValue)) {
-      this.checkedIndex = -1;
+    const checkedElement = this.formElements[/** @type {number} */ (this.checkedIndex)];
+    if (!this.multipleChoice && !autoselect && checkedElement) {
```

Comment:
If checkedElement does not exist, should the checkedIndex be set to -1 as well? That edge case doesn't seem covered by the logic here, but I could be wrong as no tests are failing.

Reply:
I believe if checkedElement doesn't exist, checkedIndex should already be set to -1. I don't think that's the place where it should set ?

Reply:
Should indeed be the case that checkedEl will only not be found if index is already -1